src/SFA.DAS.Commitments.Api.IntegrationTests/DatabaseSetup/Generators/DataLockStatusGenerator.cs

```diff
@@ -0,0 +1,192 @@
+            //new TrainingCourse { FworkCode = "548", ProgType = "3", PwayCode = "1" },
+            //new TrainingCourse { FworkCode = "418", ProgType = "2", PwayCode = "1" },
+            //new TrainingCourse { ProgType = "25", StandardCode = "6" },
+
+            //return trainingType == TrainingType.Standard
+            //    ? $"{dataLockEvent.IlrStandardCode}" :
+            //    $"{dataLockEvent.IlrFrameworkCode}-{dataLockEvent.IlrProgrammeType}-{dataLockEvent.IlrPathwayCode}";
+
+            return trainingType == TrainingType.Standard
+                ? "6"
+                : ExampleFrameworkCourseCodes[Random.Next(ExampleFrameworkCourseCodes.Length)];
+        }
+
+        private DateTime GenerateDataLockEventDatetime()
+        {
+            return SystemStartDate + new TimeSpan(0, 0, Random.Next(SecondsSinceSystemStartDate));
+        }
+
+        private IEnumerable<DateTime> GenerateStartDates(int count)
+        {
+            // case 0 and 1 aren't necessary, they're just optimisations
+            switch (count)
+            {
+                case 0:
+                    return Enumerable.Empty<DateTime>();
+                case 1:
+                    return Enumerable.Repeat(GenerateStartDate(), 1);
+                default:
+                    return Enumerable.Range(0, MonthsSinceSystemStartDate + 6).OrderBy(m => Random.Next()).Take(count)
+                        .Select(m => SystemStartDate.AddMonths(m)).OrderBy(d => d);
+            }
+        }
+
+        private DateTime GenerateStartDate()
+        {
+            return SystemStartDate.AddMonths(Random.Next(0, MonthsSinceSystemStartDate+6));
+        }
+
+        private DateTime? GenerateIlrPriceEffectiveToDate(DateTime? ilrEffectiveFromDate)
+        {
+            return ilrEffectiveFromDate?.AddMonths(3 + Random.Next(9));
+        }
+
+        private decimal? GenerateTotalCost()
+        {
+            return 200+(Random.Next(300)*10);
+        }
+
+        private TriageStatus GenerateTriageStatus(DataLockErrorCode errorCode)
+        {
+            if (errorCode == DataLockErrorCode.None)
+                return TriageStatus.Unknown;
+
+            // if errorcode is one of the 4 change codes
+            if ((errorCode &
+                 (DataLockErrorCode.Dlock03 | DataLockErrorCode.Dlock04 | DataLockErrorCode.Dlock05 | DataLockErrorCode.Dlock06)) != 0)
+            {
+                return Random.Next(2) == 0 ? TriageStatus.Restart : TriageStatus.Unknown;
+            }
+
+            if ((errorCode & (DataLockErrorCode.Dlock07 | DataLockErrorCode.Dlock09)) != 0)
+            {
+                return Random.Next(2) == 0 ? TriageStatus.Change : TriageStatus.Unknown;
+            }
+
+            // FixInIlr is not currently used
+            return TriageStatus.Unknown;
+        }
+
+        private bool GenerateIsResolved(TriageStatus triageStatus)
+        {
+            return triageStatus == TriageStatus.Unknown ? false : Random.Next(2) == 0;
```

Comment:
can be rewritten:

'return triageStatus != TriageStatus.Unknown && Random.Next(2) == 0;'

Reply:
The Resharper suggestion made me consider this, but I thought the expression as is, is more descriptive of what the code is trying to achieve, i.e. If the triage status is unknown then it isn't resolved, otherwise it's random.
Whereas, the simplification reads more like 'if triage status isn't unknown and random' which IMO isn't as clear to its purpose.
Especially as it's test code, I preferred the more descriptive version over the micro-optimised version.